aim/tools/cli/commands/manager.py

```diff
@@ -248,12 +274,42 @@ def load_domains(ctx, replace, enforce):
                           for x in manager.find(aim_ctx,
                                                 resource.PhysicalDomain)]
             all_epgs = manager.find(aim_ctx, resource.EndpointGroup)
+            if mappings:
+                host_mappings = {}
+                # We use just the wildcard mappings, since we aren't asking
+                # APIC for host-to-EPG associations
+                db_mappings = manager.find(aim_ctx, infra.HostDomainMappingV2,
```

Comment:
I'm a little confused about this block: do_mappings() earlier in the function creates wildcarded entries for all resource.VMMDomain and resource.PhysicalDomain objects. And then this block queries the domains associated with all wildcarded entries. Won't the result be the same as what the existing code was doing (lines 271-275)?

Reply:
I'm confused here also......

Reply:
It is confusing :-(. I probably should explain this better. If the user doesn't use replace, then this will create entries from the domains it finds, but there *could* also be entries that the user had created previously in the HostDomainMappingV2 table (i.e. from running the aimctl manager host-domain-mapping-v2-create). Would additional commenting be helpful?

Reply:
I think some comment here would definitely help!

Reply:
OK. Say if the user had previously created non-wildcard entries in HostDomainMappingV2, then you are not considering those at all, and you will end up removing the association between existing EPGs and those domains, right? I think the correct thing to do here is *append* to the list of EPG-domains associations instead of replacing that list.

Reply:
Good point.

The 90% use case won't involve the --replace (probably only needed if a domain is no longer needed and must be removed, in which case the user will also want the host mapping removed, so they will have to know to remove it explicitly). It's mainly for folks who want to define an initial set of domains for an openstack instance, and then possibly to add to it. If there are host-specific mappings, then they will have to add those using the separate command anyway, so I think I agree with your points. Will change.

Reply:
I just realized that the --enforce is a bit more complicated. I think it really only can be useful when there are only wildcard mappings. If you have a mix of host-specific and wildcard mappings, you can't make associations because you have no information about hosts (neutron has port-to-host mapping information -- you have no way to know whether a host-specific domain mapping or a wildcard domain mapping should be used). My thinking here is that the enforce option would ignore host-specific mappings and assume that a new set of mappings is being done, using only wildcard mappings.

Reply:
May be --enforce should error out when there are host-specific mappings instead of potentially breaking existing EPGs?

Reply:
Yeah -- either that, or I could check if the EPG already has a domain, and if there is a host-specific mapping with that domain, then don't update that EPG. That said, I think the erroring out makes more sense, and would cover the 90% use case workflow. The host-specific mapping can only be associated through port-binding, so it doesn't make sense to try to do something clever here, IMHO.